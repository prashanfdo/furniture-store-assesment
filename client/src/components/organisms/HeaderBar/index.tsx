import SearchBox from "components/organisms/SearchBox";
import { Link } from "react-router-dom";

export default function HeaderBar() {
  return (
    <>
      <header className="fixed w-full h-16 left-0 top-0 bg-white md:px-6 md:py-3 grid grid-cols-[6rem_1fr] !pl-14 lg:!pl-6">
        <Link to="/" className="grid w-24 place-items-center">
          <img className="h-full" src="/home24_logo.svg" alt="site logo" />
        </Link>
        <div className="grid mx-4 place-items-center lg:mr-24">
          <SearchBox />
        </div>
      </header>
    </>
  );
}
