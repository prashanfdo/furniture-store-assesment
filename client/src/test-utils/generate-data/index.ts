export * from "./article";
export * from "./category";
export * from "./search";
