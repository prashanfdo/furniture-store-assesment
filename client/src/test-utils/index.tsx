import { render, RenderOptions, renderHook, RenderHookOptions, RenderHookResult } from "@testing-library/react";
import fs from "fs";
import React, { FC, ReactElement } from "react";

import user from "@testing-library/user-event";
import { ProvidersForTests } from "providers";

type GraphQlMocks = React.ComponentProps<typeof ProvidersForTests>["mocks"];

type WrapperProps = React.ComponentProps<typeof ProvidersForTests> & {
  children: React.ReactNode;
};
const Wrapper: FC<WrapperProps> = ({ children, mocks, initialRoute }: WrapperProps) => {
  return (
    <ProvidersForTests mocks={mocks} initialRoute={initialRoute}>
      {children}
    </ProvidersForTests>
  );
};
const wrapper =
  (mocks: GraphQlMocks, initialRoute?: string) =>
  ({ children }: { children: React.ReactNode }) =>
    (
      <Wrapper mocks={mocks} initialRoute={initialRoute}>
        {children}
      </Wrapper>
    );

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper"> & { gqlMocks: GraphQlMocks; initialRoute: string },
) => {
  const view = render(ui, { wrapper: wrapper(options?.gqlMocks, options?.initialRoute), ...options });
  const style = document.createElement("style");
  style.innerHTML = fs.readFileSync("src/index.out.css", "utf8");
  document.head.appendChild(style);

  return view;
};

export const customRenderHook = <T, R>(
  callback: (props: T) => R,
  options?: Omit<RenderHookOptions<T>, "wrapper"> & { gqlMocks?: GraphQlMocks; initialRoute?: string },
): RenderHookResult<R, T> => {
  return renderHook(callback, { ...options, wrapper: wrapper(options?.gqlMocks, options?.initialRoute) });
};

export { faker } from "@faker-js/faker";
export * from "@testing-library/react";
export * from "./generate-data";
export { customRender as render };
export { customRenderHook as renderHook };
export { user };
