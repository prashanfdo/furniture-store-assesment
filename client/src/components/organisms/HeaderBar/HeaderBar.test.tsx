import { render, screen } from "test-utils";
import HeaderBar from "./index";

jest.mock("components/organisms/SearchBox", () => () => <div data-testid="search-box" />);

describe("HeaderBar", () => {
  it("should render", () => {
    const { container } = render(<HeaderBar />);
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId("search-box")).toBeInTheDocument();
    expect(screen.getByRole("link")).toHaveAttribute("href", "/");
    expect(screen.getByRole("img")).toHaveAttribute("alt", "site logo");
  });
});
