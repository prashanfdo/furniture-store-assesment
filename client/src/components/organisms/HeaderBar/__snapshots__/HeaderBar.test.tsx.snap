// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar should render 1`] = `
<div>
  <header
    class="fixed w-full h-16 left-0 top-0 bg-white md:px-6 md:py-3 grid grid-cols-[6rem_1fr] !pl-14 lg:!pl-6"
  >
    <a
      class="grid w-24 place-items-center"
      href="/"
    >
      <img
        alt="site logo"
        class="h-full"
        src="/site_logo.svg"
      />
    </a>
    <div
      class="grid mx-4 place-items-center lg:mr-24"
    >
      <div
        data-testid="search-box"
      />
    </div>
  </header>
</div>
`;
